desc/magnetic_fields/_dommaschk.py

```diff
@@ -412,8 +417,6 @@ def D_m_n(R, Z, m, n):
     def body_fun(k, val):
         coef = CD_m_k(R, m, k) / gamma(n - 2 * k + 1)
         exp = n - 2 * k
-        # derivative of 0**0 is ill defined, so we do this to enforce it being 0
-        exp = jnp.where((Z == 0) & (exp == 0), 1, exp)
```

Comment:
it feels weird that this works without this fix. This should return nan when differentiated:

```python
@jit
def foo(x, p):
    return x**p

jax.jacrev(foo)(0.0, 0)
# > Array(nan, dtype=float64, weak_type=True)
```

Reply:
Ahh maybe this was giving the wrong derivative too. 

$x^0 = 0~ \forall~ x$, so $\partial_x x^0 = 0$ but this was making the derivative 1:

```python
@jit
def foo(x, p):
    p = jnp.where((x == 0) & (p == 0), 1, p)
    return x**p

jax.jacrev(foo)(0.0, 0)
# > Array(1., dtype=float64, weak_type=True)
```

I think we need a double where trick to enforce that it should be 0 with 0 derivative:
```python
@jit
def foo(x, p):
    mask = (x == 0) & (p == 0)
    p = jnp.where(mask, 1, p)
    return jnp.where(mask, 0, x**p)

jax.jacrev(foo)(0.0, 0)
# > Array(0., dtype=float64, weak_type=True)
```

Reply:
that's if p is hard-coded, but in our case its still a traced value because it changes in a loop. I added that logic because I was getting nan values and gradients without it, so I think something is probably needed